app/helpers/elasticsearch_helper.rb

```diff
@@ -30,15 +30,19 @@ def taxon_search(query, tax_levels = TaxonCount::NAME_2_LEVEL.keys, filters = {}
       search_params = {
         size: ElasticsearchHelper::MAX_SEARCH_RESULTS,
         query: {
-          query_string: {
-            query: "*#{query}*",
-            fields: ["#{level}_name"],
+          match: {
```

Comment:
I would like to see your comment in the code here:

"query by match. This type of query matches multiple strings in a fuzzy way according to the operator specified (and in this case). match is the query type recommended by ElasticSearch for full-text search."

Plus: "optimized with feedback from users" -- that's the really expensive part!

Reply:
Also it would be nice to know how `MAX_SEARCH_RESULTS` was determined.

Reply:
> optimized with feedback from users

I don't think of it as optimized, but as a fix. The previous query was just not adequate for the use case and provided seemingly wrong results.

Reply:
Added a few extra comments.